ReactSkia/components/RSkComponentScrollView.cpp

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#include "include/core/SkPaint.h"
+#include "include/core/SkPictureRecorder.h"
+
+#include "compositor/layers/PictureLayer.h"
+
+#include "ReactSkia/components/RSkComponentScrollView.h"
+#include "ReactSkia/utils/RnsUtils.h"
+#include "ReactSkia/views/common/RSkDrawUtils.h"
+
+namespace facebook {
+namespace react {
+
+#define SCROLL_NEXT 1
+#define SCROLL_PREV 0
+
+using namespace RSkDrawUtils;
+
+RSkComponentScrollView::RSkComponentScrollView(const ShadowView &shadowView , RSkSurfaceWindow * surface)
+    : RSkComponent(shadowView),
+      containerSize_(shadowView.layoutMetrics.frame.size),
+      surface_(surface) {}
+
+
+void RSkComponentScrollView::updateComponentProps(
+    const ShadowView &newShadowView,
+    bool forceUpadate) {
+
+  auto const &scrollViewProps = *std::static_pointer_cast<ScrollViewProps const>(newShadowView.props);
+
+  isScrollEnabled_ = scrollViewProps.scrollEnabled;
+
+  if( forceUpadate || (contentInset_ != scrollViewProps.contentInset)) {
+       contentInset_ = scrollViewProps.contentInset;
+       forceBitmapReset_ = true;
+  }
+
+}
+
+inline void RSkComponentScrollView::updateContainerSize() {
+
+   /* Recalculate container size,as mount/umount can happen wrt any index */
+   /* TODO Check how we can optimize this call from mount/unmount
+    *      as every instruction we have to traverse full child list */
+
+   Size frameSize = getComponentData().layoutMetrics.frame.size;
+   if(childList_.size()) {
+      for (auto &child : childList_) {
+         Rect childFrame = child->getComponentData().layoutMetrics.frame;
+
+         /* Consider the last element in y direction to get the container height */
+         if((childFrame.origin.y + childFrame.size.height) > frameSize.height)
```

Comment:
Use Rect getMaxX or getMaxY functions instead of (childFrame.origin.y + childFrame.size.height)